core/store.go

```diff
@@ -238,6 +240,37 @@ func Rename(sourceKey, destKey string) bool {
 	}, WithStoreLock(), WithKeypoolLock())
 }
 
+// TODO, Key expiry should be handled for the RandomKey command
+// with the below implementations key expiry checks will be difficult(in Redis expiration is checked with maxTries of 100)
+// this should be considered on future data structure refactoring
+func RandomKey() string {
+	randomKey := ""
+	withLocks(func() {
+		if len(keypool) == 0 {
+			return
+		}
+		idxRange := len(keypool) / 2
+		if idxRange > 1024 {
+			idxRange = 1024
+		}
+
+		index, err := rand.Int(rand.Reader, big.NewInt(int64(idxRange)))
+		if err != nil {
+			return
+		}
+		RandomIndex := int(index.Int64())
+		currentIndex := 0
+		for key := range keypool {
```

Comment:
You can just iterate using 
```go
for idx, key := range keypool
```
This way you won't have to maintain a separate counter

Reply:
I don't think this will work since ranging over a map returns the key and value, not an index. Please correct me if I'm mistaken, as I'm still new to Go.

Reply:
Whoops, my bad.